### PR TITLE
fix(app): handle the input via OT_APP_DISCOVERY__CANDIDATES 

### PR DESCRIPTION
**Pull request by koji**
# Overview
The currently Connect to a robot via IP address only can handle `string[]` so if a user sets the value via OT_APP_DISCOVERY__CANDIDATES and clicks `Setup connection` in App Settings page, the app shows white-screen.
This PR converts candidates into `string[]` if its type is `string`.
Close RQA-415

# Changelog
- add code to handle OT_APP_DISCOVERY__CANDIDATES (string)
- add a condition to avoid unexpected error

# Review requests
- [ ] When run the desktop app pass `localhost` via `OT_APP_DISCOVERY__CANDIDATES`, Connect to a robot via IP address slideout shows localhost without showing white screen

# Risk assessment
low

### CODE REVIEW

File: app/src/organisms/AppSettings/ManualIpHostnameList.tsx [hunk 1]
```diff
@@ -18,35 +18,43 @@ export function ManualIpHostnameList({
   setMostRecentAddition,
   setMostRecentDiscovered,
 }: IpHostnameListProps): JSX.Element {
-  const candidates = useSelector(
-    (state: State) => getConfig(state)?.discovery.candidates ?? []
-  )
+  const candidates = useSelector((state: State) => {
+    const results = getConfig(state)?.discovery.candidates
+    return typeof results === 'string'
```
Comment (koji) [inline app/src/organisms/AppSettings/ManualIpHostnameList.tsx]: The app recognize the input via OT_APP_DISCOVERY__CANDIDATES as `string` so needed to convert string -> string[]

Comment (brenthagen) [reply app/src/organisms/AppSettings/ManualIpHostnameList.tsx]: if `DiscoveryCandidates` can be either `string` or `string[]` we should probably change that type also

Comment (brenthagen) [reply app/src/organisms/AppSettings/ManualIpHostnameList.tsx]: also i think just the line `return [].concat(results)` would work as a single line, and avoid the two ternaries

**Comment by koji** [comment]
- [ ] fix key issue

**Review by brenthagen — APPROVED** [review]

**Pull request merged**